pulp_2to3_migration/app/plugin/content.py

```diff
@@ -120,9 +122,16 @@ async def create_artifact(self, pulp2_storage_path, expected_digests={}, expecte
             artifact.size = expected_size
             return artifact
 
-        artifact = Artifact.init_and_validate(pulp2_storage_path,
-                                              expected_digests=expected_digests,
-                                              expected_size=expected_size)
+        try:
+            artifact = Artifact.init_and_validate(pulp2_storage_path,
+                                                  expected_digests=expected_digests,
+                                                  expected_size=expected_size)
+        except (DigestValidationError, FileNotFoundError, SizeValidationError):
+            if self.skip_corrupted:
+                _logger.warn(f'The content located in {pulp2_storage_path} is missing or '
+                             f'corrupted. It was skipped during Pulp 2to3 migration.')
+                return
+            raise
```

Comment:
@goosemania  Do you think it will be valuable if we would raise a more migration friendly error something along the lines " the artifact is whether missing or is corrupted please fix it before proceeding or use 'skip corrupted content option in the next run'

Reply:
yeah, can do.

Reply:
did we test the scenario when a missing file has been re-added and then migration re-run?

Reply:
added in pulp2?

Reply:
yes in pulp2

Reply:
Nope, I haven't tested that.
If the file is the only thing that changed (no changes in the timestamp), it won't be remigrated on the next run.
We need to document that!

Reply:
The reason i have asked question about the missing file is to figure out how useful 'skip corrupted' option is - if the user still cannot migrate the data even if he has fixed the missing file then maybe it would be better to just issue the warning and not raise the error.

Reply:
Tested. All good.